src/backend/distributed/executor/multi_utility.c

```diff
@@ -613,11 +613,48 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 					}
 
 					constraint = (Constraint *) command->def;
-					if (ConstraintIsAForeignKey(constraint->conname, relationId))
+					if (constraint->contype == CONSTR_FOREIGN)
 					{
 						InvalidateForeignKeyGraph();
 					}
 				}
+				else if (alterTableType == AT_AddColumn)
+				{
+					List *columnConstraints = NIL;
+					ListCell *columnConstraint = NULL;
+					Oid relationId = InvalidOid;
+					LOCKMODE lockmode = NoLock;
+
+					ColumnDef *columnDefinition = (ColumnDef *) command->def;
+					columnConstraints = columnDefinition->constraints;
+					if (columnConstraints)
+					{
+						ErrorIfUnsupportedAlterAddConstraintStmt(alterTableStatement);
+					}
+
+					lockmode = AlterTableGetLockLevel(alterTableStatement->cmds);
+					relationId = AlterTableLookupRelation(alterTableStatement, lockmode);
+					if (!OidIsValid(relationId))
+					{
+						continue;
+					}
+
+					foreach(columnConstraint, columnConstraints)
+					{
+						Constraint *constraint = (Constraint *) lfirst(columnConstraint);
+
+						if (constraint->conname == NULL &&
+								(constraint->contype == CONSTR_PRIMARY ||
+								 constraint->contype == CONSTR_UNIQUE ||
+								 constraint->contype == CONSTR_FOREIGN))
+						{
+							ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+											errmsg("cannot execute ADD COLUMN command with "
+												   "PRIMARY KEY, UNIQUE and FOREIGN constraints"),
+											errhint("You can issue each command separately")));
```

Comment:
I don't think it's clear from the hint what you need to do. 

Maybe "Add the column without a constraint and then run an ALTER TABLE .. ADD CONSTRAINT command"

We could also add some more context, e.g. `errdetail("Adding a column with a constraint in one command is not supported because all constraints in Citus must have explicit names")`